### PR TITLE
ZER-76 - Add view user page on Admin side

### PR DESCRIPTION
**Pull request by halyna-m**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by andriibureviy — DISMISSED** [review]

**Review by Nazarii1017 — APPROVED** [review]
all good

File: app/views/admins/users/index.html.slim [hunk 1]
```diff
@@ -7,7 +7,9 @@ div class="container home"
           tr
             td = t '.table.email_col'
             td = t '.table.last_visit_col'
+            td = t '.table.view_info_col'
         - @users.each do |user|
           tr
             td = user.email
             td = user.last_sign_in_at
+            td = link_to (fa_icon "eye"), admins_user_path(user.id), id: "link-to"
```
Comment (maksymst) [inline app/views/admins/users/index.html.slim:15]:
```suggestion
            td = link_to fa_icon('eye'), admins_user_path(user.id), id: 'link-to'
```

File: app/views/admins/users/show.html.slim [hunk 2]
```diff
@@ -0,0 +1,24 @@
+div class="container"
+  h2 = t '.header'
+  ul
+    li
+        span = t '.table.email_col'
+        span = @user.email
```
Comment (maksymst) [inline app/views/admins/users/show.html.slim:6]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

File: config/locales/en.yml [hunk 3]
```diff
@@ -57,6 +57,17 @@ en:
...
+      show: 
+        header: "User info"
+        table: 
+          email_col: "Email: "
+          first_name_col: "First name: "
+          last_name_col: "Last name: "
+          country_col: "Country: "
+          last_sign_in_col: "Last sign in date and time: "
+          current_IP_col: "Current sign in IP: "
+          last_IP_col: "Last sign in IP: "
```
Comment (maksymst) [inline config/locales/en.yml:70]: avoid using empty spaces

File: spec/features/admin/users_spec.rb [hunk 4]
```diff
@@ -1,12 +1,41 @@
...
+      expect(page).to have_content 'Country'
+      expect(page).to have_content 'Last sign in date and time'
+      expect(page).to have_content 'Current sign in IP'
+      expect(page).to have_content 'Last sign in IP'
+    end
+  end
+end
+
+describe 'user info page' do
+  context 'when visit user page with id bigger than amount of users' do
```
Comment (maksymst) [inline spec/features/admin/users_spec.rb:35]:
```suggestion
  context 'when viewing non-existing user' do
```

File: app/views/admins/users/index.html.slim (hunk 1 above)
Comment (maksymst) [inline app/views/admins/users/index.html.slim:15]: adding the same id within a loop will render invalid HTML. I can have only one unique id per page. Assign an id to a `tr` element, that includes `user.id`

File: config/routes.rb [hunk 5]
```diff
@@ -5,7 +5,7 @@
   devise_for :users, controllers: { registrations: 'users/registrations' }
   resources :calculators, only: [:show]
   namespace :admins do
-    resources :users, only: [:index]
+    resources :users, only: [:index, :show]
```
Comment (maksymst) [inline config/routes.rb:8]: https://github.com/rubocop/ruby-style-guide#percent-i

File: spec/features/admin/users_spec.rb [hunk 6]
```diff
@@ -1,12 +1,41 @@
...
+    expect(page).to have_content time_login
+  end
+
+  context 'when user clicks show icon' do
+    it 'redirects to user info page' do
       visit '/admins/users'
-      expect(page).to have_content 'test1@gmail.com'
-      expect(page).to have_content time_login
+      page.find("[id='link-to']").click
+      expect(page).to have_current_path('/admins/users/1')
```
Comment (maksymst) [inline spec/features/admin/users_spec.rb:22]: record identifiers in tests should not be static

**Review by maksymst — CHANGES_REQUESTED** [review]

**Review by andriibureviy — APPROVED** [review]

**Review by Nazarii1017 — APPROVED** [review]
all good

File: app/views/admins/users/index.html.slim [hunk 7]
```diff
@@ -7,7 +7,9 @@ div class="container home"
           tr
             td = t '.table.email_col'
             td = t '.table.last_visit_col'
+            td = t '.table.view_info_col'
         - @users.each do |user|
-          tr
+          tr id="user_page_#{user.id}"
```
Comment (maksymst) [inline app/views/admins/users/index.html.slim:12]:
```suggestion
          tr id="user-info-#{user.id}"
```

**Review by maksymst — APPROVED** [review]

**Review by andriibureviy — APPROVED** [review]

**Review by Nazarii1017 — APPROVED** [review]

**Pull request merged**